Introduce dock bar for left panel

aglinxinyuan opened the pull request:
In this PR:
1. Introduce a dock bar for the left panel
2. Move the version history entry from the top menu bar timestamp to the left panel dock bar.

We will extend the left panel menu bar in future PRs since the other functionalities require dependencies.

<img width="264" alt="Screenshot 2024-01-17 at 12 06 46 AM" src="https://github.com/Texera/texera/assets/11544314/9ac195a8-7c51-4a7e-bce8-019bddcad1ca">

Yicong-Huang reviewed (approved):
Please change it to `Dock` and related names instead of menu. Fix the "move" panel operation so that user can drag the title to move the panel.

aglinxinyuan commented:
> Please change it to `Dock` and related names instead of menu. Fix the "move" panel operation so that user can drag the title to move the panel.

Updated and fixed.
